frontend/src/NewBaby.tsx

```diff
@@ -0,0 +1,154 @@
+import React, {FormEvent} from 'react';
+import axios from "axios";
+import {Link, useNavigate} from "react-router-dom";
+import styled from "styled-components";
+
+type NewBabyProps = {
+    getAllBabies: () => void
+}
+
+export default function NewBaby(props: NewBabyProps) {
+    const navigate = useNavigate();
+
+    const [newBaby, setNewBaby] = React.useState(
+        {
+            name: "",
+            birthday: "",
+            weight: "",
+            height: "",
+            gender: "",
+        });
+
+    const baseUrl = '/api/baby';
+
+    const postNewBaby = (event: FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+        axios.post(baseUrl, newBaby)
+            .then(function (response) {
+                console.log(response);
+            })
+            .then(props.getAllBabies)
+            .then(() => navigate("/babyoverview"))
+            .catch((e) => console.log("POST ERROR: " + e))
+        setNewBaby({name: "", birthday: "", weight: "", height: "", gender: ""});
```

Comment:
Would put this in a then() just to be sure it is synchronized the the post-request

Reply:
setNewBaby is not a content from the axios function